HW04.Rmd

```diff
@@ -29,35 +48,76 @@ Knitr is part of the knitr package. Allows you to change column names and add ca
 
 
 ```{r month, warning=FALSE}
+gun_deaths_month <- gun_deaths %>%
+    group_by(month) %>%
+    count(month) %>%
+    rename(number = n)
 
+kable(gun_deaths_month, format = "markdown")
 ```
 
 
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
 ```{r month_chart}
-
+# create human-readable label for each month
+gun_deaths$month <- factor(gun_deaths$month, labels = c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"))
+
+# Draw the bar chart
+ggplot(gun_deaths, aes(x = month, fill = month)) +
+    geom_bar() + 
+    xlab("\nMonth") + 
+    ylab("Number of Death\n") +
+    scale_y_continuous(breaks=seq(0, 9000, by = 1000))
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
 ```{r intent}
-
+ggplot(gun_deaths, aes(x=fct_infreq(intent), fill=intent)) +
+    geom_bar() +
+    xlab("\nType of Intent") + 
+    ylab("Number of Death\n") +
+    scale_y_continuous(breaks=seq(0, 70000, by = 10000))
 ```
 
 ## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
 
 ```{r age_&_sex}
-
+# calculate average age by sex
+avg_age <- gun_deaths %>%
+    group_by(sex) %>%
+    summarize(mean_age=mean(age, na.rm = TRUE)) 
+
+# round to the nearest tenth
+avg_age$mean_age <- round(avg_age$mean_age, digits = 2)
+
+# draw boxplot by sex and print average age
+ggplot(gun_deaths, aes(x = age, y = sex)) +
+    geom_boxplot() +
+    xlab("\nAge") + 
+    ylab("Sex\n") +
+    scale_x_continuous(breaks=seq(0, 100, by = 10)) + 
+    stat_summary(fun = mean, colour = "red", geom = "point", shape = 18, size = 2) +
+    geom_text_repel(avg_age, mapping = aes(x = mean_age, y = sex, label = mean_age), inherit.aes = FALSE)
 ```
 
 
 ## How many white males with at least a high school education were killed by guns in 2012?
 
 ```{r education}
+# convert education to a factor 
+gun_deaths$education <- factor(gun_deaths$education)
 
-```
+# get a summary of education 
+summary(gun_deaths$education)
 
+# use filter to select the records we are interested in
+gun_deaths %>%
+    filter(race == "White" & education == c("HS/GED", "Some college", "BA+") & year == 2012) %>%

```

Comment:
I got a different number here (15199). I wonder is it because for education you used "==" instead of "%n%"?